### PR TITLE
Add ApiGovFrance.php to Record Collectors.

### PR DESCRIPTION
**Pull request by rembiesa**
image

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #16256 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/16256)

File: app/RecordCollectors/ApiGovFrance.php [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		],
+		'vatNumber' => [
+			'labelModule' => '_Base',
+			'label' => 'VAT',
+			'typeofdata' => 'V~O',
+		],
+	];
+
+	/** {@inheritdoc} */
+	protected $modulesFieldsMap = [
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/ApiGovFrance.php]: Leads ... ??

File: app/RecordCollectors/ApiGovFrance.php [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+	/** {@inheritdoc} */
+	public function search(): array
+	{
+		if (!$this->isActive()) {
+			return [];
+		}
+		$vatNumber = str_replace([' ', ',', '.', '-'], '', $this->request->getByType('vatNumber', 'Text'));
+		$sicCode = str_replace([' ', ',', '.', '-'], '', $this->request->getByType('sicCode', 'Text'));
+		$companyName = str_replace([' ', ',', '.', '-'], '', $this->request->getByType('companyName', 'Text'));
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/ApiGovFrance.php]: ?

File: app/RecordCollectors/ApiGovFrance.php [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	];
+
+	/** @var string CH sever address */
+	private $url = 'https://recherche-entreprises.api.gouv.fr/';
+
+	/** @var int count of elements for page */
+	const PER_PAGE = 5;
+
+	/** @var int number of page */
+	const PAGE = 1;
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/ApiGovFrance.php]: ?

File: app/RecordCollectors/ApiGovFrance.php [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		if(empty($vatNumber) && empty($sicCode) && empty($companyName)) {
+			return [];
+		}
+
+		if (!empty($vatNumber)) {
+			$query['q'] = $vatNumber;
+		} elseif (!empty($companyName) && empty($vatNumber)) {
+			$query['q'] = $companyName;
+		}
+		$query['page'] = self::PAGE;
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/ApiGovFrance.php]: ?

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

**Pull request merged**